Remove compute_uv flag from torch.linalg.svd

IvanYashchuk opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57181 Change name for namedtuple return of torch.linalg.svd
* **#57180 Remove compute_uv flag from torch.linalg.svd**

We have now a separate function for computing only the singular values.
`compute_uv` argument is not needed and it was decided in the
offline discussion to remove it. This is a BC-breaking change but our
linalg module is beta, therefore we can do it without a deprecation
notice.

Differential Revision: [D28142163](https://our.internmc.facebook.com/intern/diff/D28142163)

IvanYashchuk committed:
Remove compute_uv flag from torch.linalg.svd

We have now a separate function for computing only the singular values.
`compute_uv` argument is not needed and it was decided in the
offline discussion to remove it. This is a BC-breaking change but our
linalg module is beta, therefore we can do it without a deprecation
notice.

[ghstack-poisoned]

ezyang commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -8807,10 +8807,10 @@
   python_module: linalg
 
-- func: linalg_svd(Tensor self, bool full_matrices=True, bool compute_uv=True) -> (Tensor U, Tensor S, Tensor V)
+- func: linalg_svd(Tensor self, bool full_matrices=True) -> (Tensor U, Tensor S, Tensor V)
```
Not for this patch, but I suspeeeect we can make this structured https://github.com/pytorch/rfcs/blob/rfc-0005/RFC-0005-structured-kernel-definitions.md since I recently added support for multi output

mruberry commented:
Note: I've identified one internal usage of compute_uv=False; I've contacted the engineer to see how disruptive this change will be

mruberry commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -8807,10 +8807,10 @@
     CPU, CUDA: linalg_vector_norm_out
 
-- func: linalg_svd.U(Tensor self, bool full_matrices=True, bool compute_uv=True, *, Tensor(a!) U, Tensor(b!) S, Tensor(c!) V) -> (Tensor(a!) U, Tensor(b!) S, Tensor(c!) V)
+- func: linalg_svd.U(Tensor self, bool full_matrices=True, *, Tensor(a!) U, Tensor(b!) S, Tensor(c!) V) -> (Tensor(a!) U, Tensor(b!) S, Tensor(c!) V)
```
These changes would break any torchscripted model using torch.linalg.svd, so I'll need to validate that's not an issue internally

a solution would be to add back compute_uv without documenting it and go through a more complete deprecation process

IvanYashchuk replied:
Is there any news on that?
`torch.svd` still has the `compute_uv` argument, if that's needed for some reason, and we have `torch.linalg.svdvals` now.

mruberry replied:
I contacted an internal team and they'd like svdvals to be differentiable so they can switch to it, which seems like a reasonable request? From a project perspective it looks like that's happening soon, too, so we shouldn't be held up satisfying them

mruberry reviewed (approved):
Cool!

Interestingly SciPy also has compute_uv, even though it has svdvals(), too. In Julia, however, it looks like svd() and svdvals() are properly separated.

I'll need to verify this won't bork anyone internally before we land this

IvanYashchuk committed:
Update on "Remove compute_uv flag from torch.linalg.svd"


We have now a separate function for computing only the singular values.
`compute_uv` argument is not needed and it was decided in the
offline discussion to remove it. This is a BC-breaking change but our
linalg module is beta, therefore we can do it without a deprecation
notice.

Differential Revision: [D28142163](https://our.internmc.facebook.com/intern/diff/D28142163)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28142163).

IvanYashchuk committed:
Update on "Remove compute_uv flag from torch.linalg.svd"


We have now a separate function for computing only the singular values.
`compute_uv` argument is not needed and it was decided in the
offline discussion to remove it. This is a BC-breaking change but our
linalg module is beta, therefore we can do it without a deprecation
notice.

Differential Revision: [D28142163](https://our.internmc.facebook.com/intern/diff/D28142163)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28142163).
